radixdlt-core/radixdlt/src/test/java/com/radixdlt/statecomputer/radixengine/AbstractRadixEngineTest.java

```diff
@@ -0,0 +1,166 @@
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.statecomputer.radixengine;
+
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.Scopes;
+import com.google.inject.multibindings.Multibinder;
+import com.radixdlt.SingleNodeAndPeersDeterministicNetworkModule;
+import com.radixdlt.api.core.reconstruction.BerkeleyRecoverableProcessedTxnStore;
+import com.radixdlt.application.system.FeeTable;
+import com.radixdlt.application.tokens.Amount;
+import com.radixdlt.application.validators.state.ValidatorRegisteredCopy;
+import com.radixdlt.consensus.bft.Self;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.environment.deterministic.DeterministicProcessor;
+import com.radixdlt.mempool.MempoolConfig;
+import com.radixdlt.network.p2p.P2PConfig;
+import com.radixdlt.network.p2p.RadixNodeUri;
+import com.radixdlt.network.p2p.addressbook.AddressBook;
+import com.radixdlt.network.p2p.addressbook.AddressBookPersistence;
+import com.radixdlt.networks.NetworkId;
+import com.radixdlt.properties.RuntimeProperties;
+import com.radixdlt.statecomputer.checkpoint.MockedGenesisModule;
+import com.radixdlt.statecomputer.forks.ForksModule;
+import com.radixdlt.statecomputer.forks.MainnetForkConfigsModule;
+import com.radixdlt.statecomputer.forks.RERulesConfig;
+import com.radixdlt.statecomputer.forks.RadixEngineForksLatestOnlyModule;
+import com.radixdlt.store.DatabaseLocation;
+import com.radixdlt.store.berkeley.BerkeleyAdditionalStore;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.util.Map;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TemporaryFolder;
+
+public abstract class AbstractRadixEngineTest {
```

Comment:
I slightly prefer naming `_Base` instead if `Abstract_` (eg `RadixEngineTestBase`) because it the latter sounds like a test of an AbstractRadixEngine.

Reply:
Using `Abstract` prefix is a quite traditional approach for the Java classes naming, especially for classes declared as abstract (which is the case).

Reply:
I've seen both conventions, but in this code base, we have two `_Base` and one `Abstract_` in the codebase already. I feel `_Base` is clearer about the intention of the class, and as well as the unclear semantics in a Test class (as per my previous comment), I also feel Abstract is a slightly weird abstraction leakage - it describes a property of the class, instead of what it's used for (and reminds me of the AbstractComponentFactoryInstantiatorBean java joke xD), but I'm ultimately happy with either.

Reply:
For me, I get `RadixApiBase` and `LatencyProviderBase` vs `AbstractHttpServer` (which isn't even abstract and no longer exists, I only have an old class file)

Reply:
```
siy@bobryzen:~/Projects/Radix-quick-look/radixdlt$ find . -name "*Base*"
./radixdlt-java-common/build/classes/java/main/com/radixdlt/utils/Base58.class
./radixdlt-java-common/src/main/java/com/radixdlt/utils/Base58.java
./radixdlt-core/radixdlt/build/classes/java/main/com/radixdlt/qualifier/LatencyProviderBase.class
./radixdlt-core/radixdlt/src/main/java/com/radixdlt/qualifier/LatencyProviderBase.java
```
The `LatencyProviderBase` is an annotation and does not serve as a base class for other classes.